client/src/components/Dashboard.test.js

```diff
@@ -29,7 +29,7 @@ describe('<Dashboard />', () => {
     fireEvent.click(button); getByText('Balls: 0');
```

Comment:
this is a great pattern for testing this behavior--verifies the expected UI at every step the user can take

Reply:
that being said, since you didn't catch the weird behavior existing in the app--ball 4 resets the count, but also increments the _inning_ unexpectedly, maybe you should also perform a counter check for the other displayed values at the end of the ball 4 simulation